src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -517,25 +517,34 @@ public object Dequeue(ThreadPoolWorkQueueThreadLocals tl, ref bool missedSteal)
             return callback;
         }
 
+        /// <summary>
+        /// Dispatches work items to this thread.
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if this thread did as much work as was available or its quantum expired.
+        /// <c>false</c> if this thread stopped working early.
+        /// </returns>
         internal static bool Dispatch()
         {
             ThreadPoolWorkQueue outerWorkQueue = ThreadPoolGlobals.workQueue;
+
             //
-            // The clock is ticking!  We have ThreadPoolGlobals.TP_QUANTUM milliseconds to get some work done, and then
-            // we need to return to the VM.
+            // Save the start time
             //
-            int quantumStartTime = Environment.TickCount;
+            int startTickCount = Environment.TickCount;
 
             //
             // Update our records to indicate that an outstanding request for a thread has now been fulfilled.
             // From this point on, we are responsible for requesting another thread if we stop working for any
             // reason, and we believe there might still be work in the queue.
             //
-            // Note that if this thread is aborted before we get a chance to request another one, the VM will
+            // CoreCLR: Note that if this thread is aborted before we get a chance to request another one, the VM will
             // record a thread request on our behalf.  So we don't need to worry about getting aborted right here.
             //
             outerWorkQueue.MarkThreadRequestSatisfied();
 
+            Interlocked.Increment(ref outerWorkQueue.numWorkingThreads);
```

Comment:
Is this number just going to go up forever?

Reply:
Looks like I accidentally left out the part that decrements it. I will check it when I get back to my computer. Thanks for spotting it!

Reply:
Fixed.